### PR TITLE
Fixed #29724 -- Fixed timezone handling in ModelAdmin.date_hierarchy queries.

### PR DESCRIPTION
**Pull request by hramezani**
Fixes [ticket 29724](https://code.djangoproject.com/ticket/29724).
Refresh [current PR](https://github.com/django/django/pull/10380).
Fixed failed tests by adding `is_dst = True` to `Trunc`.

### CODE REVIEW

File: django/db/models/query.py [hunk 1]
```diff
@@ -890,7 +890,7 @@ def datetimes(self, field_name, kind, order='ASC', tzinfo=None):
         else:
             tzinfo = None
         return self.annotate(
-            datetimefield=Trunc(field_name, kind, output_field=DateTimeField(), tzinfo=tzinfo),
+            datetimefield=Trunc(field_name, kind, output_field=DateTimeField(), tzinfo=tzinfo, is_dst=True),
```
Comment (charettes) [inline django/db/models/query.py]: This should not default to `is_dst=True` as it will be backward incompatible.

We should allow passing it as a `datetimes` kwarg and default to `None` just like `Trunc(is_dst=None)` does and pass `is_dst=True` from the admin call if necessary. 

This should likely be in a different commit and include docs.

Comment (hramezani) [reply django/db/models/query.py]: Thanks @charettes.
I added the `is_dst` parameter to `datetimes` in a separate commit.

**Comment by felixxm** [comment]
I moved first commit to a separate PR #12593 and added test.

**Comment by hramezani** [comment]
@felixxm Thanks for merging the first commit in separate PR.
Shall I rebase the PR?

**Comment by felixxm** [comment]
Yes, please.

**Comment by hramezani** [comment]
It's done.
